amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ShardConsumer.java

```diff
@@ -379,6 +384,11 @@ private synchronized void processData(ProcessRecordsInput input) {
     }
 
     private synchronized void executeTask(ProcessRecordsInput input) {
+        TaskExecutionListenerInput taskExecutionListenerInput = TaskExecutionListenerInput.builder()
```

Comment:
Do not share the input objects between the methods.

Reply:
As long as the fields are final sharing this way is fine.  `toBuilder` will trigger the creation of a new object.